### PR TITLE
fix gclient runhooks for windows

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

pub is pub.bat on windows

### CODE REVIEW

File: tools/run_third_party_dart.py [hunk 1]
```diff
@@ -3,8 +3,12 @@
...
 import subprocess
+import os
 
 if __name__ == '__main__':
-    subprocess.call(["src/third_party/dart/tools/sdks/dart-sdk/bin/pub", "global", "activate", "-spath", "./src/flutter/tools/generate_package_config"])
-    subprocess.call(["src/third_party/dart/tools/sdks/dart-sdk/bin/pub", "global", "run", "generate_package_config:generate_from_legacy", "src/flutter/flutter_frontend_server/.packages"])
-    subprocess.call(["src/third_party/dart/tools/sdks/dart-sdk/bin/pub", "global", "run", "generate_package_config:generate_from_legacy", "src/flutter/tools/const_finder/.packages"])
+    pub = "src/third_party/dart/tools/sdks/dart-sdk/bin/pub"
+    if os.name == "nt":
+        pub = "src/third_party/dart/tools/sdks/dart-sdk/bin/pub.bat"
```
Comment (jason-simmons) [inline tools/run_third_party_dart.py:11]: Actually I just tried this on my local Windows box and it only seems to work with backslashes, like:
`subprocess.call(["bin\\cache\\dart-sdk\\bin\\pub.bat"])`

File: tools/run_third_party_dart.py [hunk 2]
```diff
@@ -3,8 +3,12 @@
...
+    subprocess.call([pub, "global", "activate", "-spath", "./src/flutter/tools/generate_package_config"])
```
Comment (dnfield) [inline tools/run_third_party_dart.py:12]: use subprocess.check_call

File: tools/run_third_party_dart.py (hunk 1 above)
Comment (jonahwilliams) [reply tools/run_third_party_dart.py:11]: It was working for several weeks when I was using `dart pub` on windows

File: tools/run_third_party_dart.py (hunk 2 above)
Comment (jonahwilliams) [reply tools/run_third_party_dart.py]: done

File: tools/run_third_party_dart.py (hunk 1 above)
Comment (jonahwilliams) [reply tools/run_third_party_dart.py]: switched to backslashes

Comment (dnfield) [reply tools/run_third_party_dart.py]: nit: use os.path.join

Comment (dnfield) [reply tools/run_third_party_dart.py]: And then you can just conditionally define a string called `BAT` based on whether it's nt and append that to the output of path.join.

**Comment by jonahwilliams** [comment]
runhooks passed on windows, landing

**Pull request merged**